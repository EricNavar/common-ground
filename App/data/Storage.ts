import AsyncStorage from '@react-native-async-storage/async-storage';

export const Storage = {
    get: async (key: string) : Promise<string | null> => {
        return await AsyncStorage.getItem(key);
    },
    set: async (key: string, value: string) => {
        await AsyncStorage.setItem(key, value).then(() => {
            console.log('Stored ', value, ' at ', key);
        }).catch((error: any) => {
            console.log('Error storing data at ', key, error);
        });
    },
};