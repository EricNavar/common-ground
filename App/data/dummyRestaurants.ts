export const losPollosHermanos = {
    name: 'Los Pollos Hermanos',
    pic: 'https://static.independent.co.uk/s3fs-public/thumbnails/image/2015/05/01/15/lospolloshermanos.jpg?width=1200',
};

export const tacoSal = {
    name: 'Taco Sal',
    pic: 'https://static.wikia.nocookie.net/breakingbad/images/c/c0/TacoSal.png/revision/latest?cb=20191120130106',
};


export const loyolasFamilyRestaurant = {
    name: 'Loyola\'s Family Restaurant',
    pic: 'https://www.breakingbad-locations.com/wp-content/uploads/2014/09/E06-Cornered.mkv_000921613.jpg',
};
