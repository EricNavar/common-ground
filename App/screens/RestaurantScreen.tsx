import React from 'react';
import {
  Text,
  View,
  Image,
  StyleSheet,
  Button,
  ScrollView,
  TouchableOpacity,
} from 'react-native';
import { Restaurant } from '../commonTypes';
import { PriceRating } from '../components/PriceRating';
import { Review } from '../components/Review';
import { getBusinessDetails } from '../api/yelpApi';
import Icon from 'react-native-vector-icons/FontAwesome5';

const styles = StyleSheet.create({
  image: {
    width: '100%',
    height: 200,
  },
  padding: {
    paddingHorizontal: 24,
  },
  title: {
    marginBottom: 12,
    marginTop: 18,
    fontSize: 24,
    fontWeight: 'bold',
  },
  sectionTitle: {
    fontWeight: 'bold',
    marginVertical: 10,
    fontSize: 18,
    marginRight: 'auto',
    marginTop: 30,
  },
  averageReviewRating: {
    fontSize: 18,
    marginBottom: 10,
  },
  averageReview: {
    fontSize: 12,
    color: '#444444',
  },
  gallery: {
    display: 'flex',
    flexDirection: 'row',
    marginBottom: 6,
    marginHorizontal: -20,
  },
  galleryImage: {
    height: 100,
    width: 100,
    borderRadius: 2,
    marginLeft: 20,
  },
  marginTop: {
    marginTop: 12,
  },
  row: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'baseline',
  },
});

type RestaurantScreenProps = {
  navigation: {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    navigate: any;
  };
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  route: {
    params: {
      restaurant: Restaurant;
    };
  };
  restaurant: Restaurant;
};

function RestaurantScreen(props: RestaurantScreenProps) {
  const restaurant = props.route.params.restaurant;

  const [galleryPhotos, setGalleryPhotos] = React.useState<string[]>([]);

  const getGalleryPhotos = async () =>
    getBusinessDetails(restaurant.id).then((data) => {
      return data.photos;
    });

  React.useEffect(() => {
    getGalleryPhotos().then((photos) => {
      setGalleryPhotos(photos);
    });
  }, []);

  if (!restaurant) {
    return null;
  }

  const distanceFormatted = restaurant.distanceMiles
    ? Math.floor(restaurant.distanceMiles * 100) / 100
    : undefined;

  return (
    <ScrollView>
      <Image style={styles.image} source={{ uri: restaurant.thumbnail }} />
      <View style={styles.padding}>
        <Text style={styles.title}>{restaurant.name}</Text>
        <PriceRating rating={restaurant.priceRating} />
        <View style={{ display: 'flex', flexDirection: 'row' }}>
          <TouchableOpacity
            style={{
              justifyContent: 'center',
              alignContent: 'center',
              alignItems: 'center',
              marginRight: 8,
            }}
          >
            <Icon
              name="directions"
              size={16}
              color="#FF6D6E"
              style={{ alignSelf: 'center' }}
            />
            <Text style={{ fontSize: 8 }}>Directions</Text>
          </TouchableOpacity>
          {restaurant.address ? (
            <View>
              {restaurant.address.line1 ? (
                <Text style={styles.marginTop}>{restaurant.address.line1}</Text>
              ) : null}
              {restaurant.address.line2 ? (
                <Text>{restaurant.address.line2}</Text>
              ) : null}
            </View>
          ) : null}
          {restaurant.distanceMiles ? (
            <Text>{distanceFormatted} miles away</Text>
          ) : null}
        </View>
        <Text style={styles.marginTop}>{restaurant.description}</Text>
        <View style={styles.row}>
          <Text style={styles.sectionTitle}>Reviews</Text>
          {restaurant.reviews && restaurant.reviews.length > 0 ? (
            <Button
              title="See all"
              onPress={() => {
                console.log('see all');
              }}
              color="#ff6e6e"
            />
          ) : null}
        </View>
        <Text style={{ display: 'flex' }}>
          <Text style={styles.averageReviewRating}>
            {restaurant.starRating}
            <Icon name="star" size={16} color="orange" />
          </Text>
          <Text style={styles.averageReview}> average review on Yelp</Text>
        </Text>
        {restaurant.reviews ?
          restaurant.reviews.map((review, index) => (
            <Review key={index} {...review} />
          )) : null}
        {galleryPhotos && galleryPhotos.length > 0 ? (
          <View>
            <View style={styles.row}>
              <Text style={styles.sectionTitle}>Gallery</Text>
            </View>
            <ScrollView style={styles.gallery} horizontal={true}>
              {galleryPhotos.map((image: string, index: number) => (
                <Image
                  style={styles.galleryImage}
                  source={{ uri: image }}
                  key={index}
                />
              ))}
            </ScrollView>
          </View>
        ) : null}
      </View>
    </ScrollView>
  );
}

export { RestaurantScreen };
