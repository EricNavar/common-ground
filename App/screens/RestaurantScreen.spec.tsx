import { render } from '@testing-library/react-native';
import React from 'react';
import { losPollosHermanos } from '../data/dummyRestaurants';
import { RestaurantScreen } from './RestaurantScreen';

jest.mock(
    'react-native-vector-icons/FontAwesome5',
    () => 'MockedMaterialCommunityIconsicons'
);

jest.mock(
  'react-native-vector-icons/FontAwesome',
  () => 'MockedFontAwesomeIconsicons'
);

describe('ProfileScreen', () => {
  const dummyNavigation = {
    navigate: '',
  };

  it('Renders Restaurant screen correctly', () => {
    const route = {
      params: {
        restaurant: losPollosHermanos,
      },
    };
    const result = render(
      <RestaurantScreen
        restaurant={losPollosHermanos}
        navigation={dummyNavigation}
        route={route}
      />
    );

    expect(result.toJSON()).toMatchSnapshot();
  });
});
