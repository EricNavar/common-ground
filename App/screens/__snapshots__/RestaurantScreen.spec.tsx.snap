// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen Renders Restaurant screen correctly 1`] = `
<View>
  <Image
    source={
      {
        "uri": "https://static.independent.co.uk/s3fs-public/thumbnails/image/2015/05/01/15/lospolloshermanos.jpg?width=1200",
      }
    }
    style={
      {
        "height": 200,
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
          "marginVertical": 12,
        }
      }
    >
      Los Pollos Hermanos
    </Text>
    <View
      style={
        {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text>
        $
      </Text>
      <Text
        style={null}
      >
        $
      </Text>
      <Text
        style={null}
      >
        $
      </Text>
      <Text
        style={
          {
            "color": "#818181",
          }
        }
      >
        $
      </Text>
      <Text
        style={
          {
            "color": "#818181",
          }
        }
      >
        $
      </Text>
    </View>
    <Text
      style={
        {
          "marginTop": 12,
        }
      }
    >
      1515 SW 13th Street
    </Text>
    <Text>
      Gainesville, FL 32601
    </Text>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Directions
        </Text>
      </View>
    </View>
    <Text
      style={
        {
          "marginTop": 12,
        }
      }
    >
      Fast-food chain serving Mexican-inspired fare such as tacos, quesadillas & nachos.
    </Text>
    <Text
      style={
        {
          "fontSize": 18,
          "fontWeight": "bold",
          "marginVertical": 10,
        }
      }
    >
      Reviews
    </Text>
    <Text
      style={
        {
          "display": "flex",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        4
        <MockedFontAwesomeicons
          color="orange"
          name="star"
          size={16}
        />
      </Text>
      <Text
        style={
          {
            "color": "#444444",
            "fontSize": 12,
          }
        }
      >
         average review on Yelp
      </Text>
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "padding": 10,
            },
            {
              "flexBasis": 1,
              "flexShrink": 1,
              "minWidth": 50,
            },
          ]
        }
      >
        <Text>
          1
          <MockedFontAwesomeicons
            color="orange"
            name="star"
            size={16}
          />
        </Text>
      </View>
      <View
        style={
          [
            {
              "padding": 10,
            },
            {
              "flexBasis": 1,
              "flexGrow": 1,
            },
          ]
        }
      >
        <Text>
          It was ok.
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "padding": 10,
            },
            {
              "flexBasis": 1,
              "flexShrink": 1,
              "minWidth": 50,
            },
          ]
        }
      >
        <Text>
          5
          <MockedFontAwesomeicons
            color="orange"
            name="star"
            size={16}
          />
        </Text>
      </View>
      <View
        style={
          [
            {
              "padding": 10,
            },
            {
              "flexBasis": 1,
              "flexGrow": 1,
            },
          ]
        }
      >
        <Text>
          The quesalupa is good. I ordered 5 of them and they were yummy and I ate all of them.
        </Text>
      </View>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginVertical": 10,
          }
        }
      >
        Gallery
      </Text>
      <RCTScrollView
        horizontal={true}
        style={
          {
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 6,
            "marginHorizontal": -20,
          }
        }
      >
        <View>
          <Image
            source={
              {
                "uri": "https://media-cdn.tripadvisor.com/media/photo-s/1a/11/b0/75/how-can-you-possibly.jpg",
              }
            }
            style={
              {
                "borderRadius": 2,
                "height": 100,
                "marginLeft": 20,
                "width": 100,
              }
            }
          />
          <Image
            source={
              {
                "uri": "https://www.tacobell.com/images/22965_grilled_cheese_burrito_deluxe_box_750x660.jpg",
              }
            }
            style={
              {
                "borderRadius": 2,
                "height": 100,
                "marginLeft": 20,
                "width": 100,
              }
            }
          />
          <Image
            source={
              {
                "uri": "https://www.thestreet.com/.image/t_share/MTg5MjIwNjA1Nzg1MDI0NDAz/taco-bell.jpg",
              }
            }
            style={
              {
                "borderRadius": 2,
                "height": 100,
                "marginLeft": 20,
                "width": 100,
              }
            }
          />
          <Image
            source={
              {
                "uri": "https://snworksceo.imgix.net/ufa/70fcdb5c-2000-4600-a79d-f7f43e55122c.sized-1000x1000.jpg?w=1000",
              }
            }
            style={
              {
                "borderRadius": 2,
                "height": 100,
                "marginLeft": 20,
                "width": 100,
              }
            }
          />
          <Image
            source={
              {
                "uri": "https://preview.redd.it/v0jc0zl5sli41.jpg?width=3024&format=pjpg&auto=webp&s=76f0c68e71507aa323b6a97c86eb15e1e6b34809",
              }
            }
            style={
              {
                "borderRadius": 2,
                "height": 100,
                "marginLeft": 20,
                "width": 100,
              }
            }
          />
        </View>
      </RCTScrollView>
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            See all
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
